CallPlayAudio/src/main/java/com/communication/CallPlayAudio/App.java

```diff
@@ -0,0 +1,174 @@
+                    if(!identity.isEmpty()) {
+                        tasks.add(() -> {
+                            new CallPlayTerminate(callConfiguration).report(identity.trim());
+                            return true;
+                        });
+                    } else {
+                        Logger.logMessage(Logger.MessageType.ERROR, "Malformed destination identitity --> " + identity);            
+                    }
+                }
+                executorService.invokeAll(tasks);
+                executorService.shutdown();
+            }
+        } catch (Exception ex) {
+            Logger.logMessage(Logger.MessageType.ERROR, "Failed to initiate the callplayaudio call Exception -- > " + ex.getMessage());
+        }
+        deleteUser(callConfiguration.connectionString, callConfiguration.sourceIdentity);
+    }
+
+    /// <summary>
+    /// Fetch configurations from App Settings and create source identity
+    /// </summary>
+    /// <param name="appBaseUrl">The base url of the app.</param>
+    /// <returns>The <c CallConfiguration object.</returns>
+    private static CallConfiguration initiateConfiguration(String appBaseUrl) {
+        ConfigurationManager configurationManager = ConfigurationManager.getInstance();
+        String connectionString = configurationManager.getAppSettings("Connectionstring");
+        String sourcePhoneNumber = configurationManager.getAppSettings("SourcePhone");
+        String maxRetryAttemptCount = configurationManager.getAppSettings("MaxRetryCount");
+        String sourceIdentity = createUser(connectionString);
+        String audioFileName = generateCustomAudioMessage();
+        return new CallConfiguration(connectionString, sourceIdentity, sourcePhoneNumber, appBaseUrl, audioFileName, maxRetryAttemptCount);
+    }
+
+    /// <summary>
+    /// Get .wav Audio file
+    /// </summary>
+    private static String generateCustomAudioMessage() {
+        ConfigurationManager configurationManager = ConfigurationManager.getInstance();
+        String key = configurationManager.getAppSettings("CognitiveServiceKey");
+        String region = configurationManager.getAppSettings("CognitiveServiceRegion");
+        String customMessage = configurationManager.getAppSettings("CustomMessage");
+
+        try {
+            if (key != null && !key.isEmpty() && region != null && !region.isEmpty() && customMessage != null
+                    && !customMessage.isEmpty()) {
+                SpeechConfig config = SpeechConfig.fromSubscription(key, region);
+                config.setSpeechSynthesisOutputFormat(SpeechSynthesisOutputFormat.Riff16Khz16BitMonoPcm);
+
+                AudioConfig audioConfig = AudioConfig.fromWavFileInput(
+                        "src/main/java/com/communication/CallPlayAudio/audio/custom-message.wav");
+                SpeechSynthesizer synthesizer = new SpeechSynthesizer(config, audioConfig);
+                synthesizer.SpeakTextAsync(customMessage);
```

Comment:
it's understandable that you want to execute stuff while Speech is being generated, but this is not working. 
My theory is that the `synthesizer` is being killed with `synthesizer.close()` before being actually executed.
I'd recommend `SpeakText(customMessage)` or `SpeakTextAsync(customMessage).get()`